### ISSUE TITLE
Panic in naive_bayes when class not populated

### ISSUE THREAD

**Issue by rrichardson** [AtheMathmo/rusty-machine#63]
If a class column is present, but is never populated in the training set, it causes a 0 sized matrix to be created, then when calculating the variance in the distribution we get an invalid index operation. 

Since it shouldn't hurt if we just skip the iteration for that class which has 0 population, we should just be able to add a len() == 0 check here:  https://github.com/AtheMathmo/rusty-machine/blob/master/rusty-machine/src/learning/naive_bayes.rs#L165 

I'm happy to put in a PR, but figured I would wait on that since its a single line change.

**Comment by AtheMathmo** [comment]
Thanks for posting this issue!

Looking to our [good friend scikit-learn](https://github.com/scikit-learn/scikit-learn/blob/51a765a/sklearn/naive_bayes.py#L363); from my understanding it seems they only update the classes which have existing labels in the training set.

I agree this is a good approach - and also paves the way to have _partial_ training as in the scikit code. I'd be glad to accept a PR with your suggested change. Later (I think) we should adopt their approach and check the classes defined in the training set then iterate over these.
